index.html

```diff
@@ -321,107 +334,138 @@
-          PerformanceEntryList getEntriesByName (DOMString name, optional DOMString type);
-        };
-        </pre>
-        <section>
-          <h2><dfn>getEntries()</dfn> method</h2>
-          <p>Returns a <a>PerformanceEntryList</a> object returned by
-          <a href="#filter-buffer-by-name-and-type"></a> algorithm with
-          <var>buffer</var> set to <a>observer buffer</a>, and
-          <var>name</var> and <var>type</var> set to `null`.</p>
-        </section>
-        <section>
-          <h2><dfn>getEntriesByType()</dfn> method</h2>
-          <p>Returns a <a>PerformanceEntryList</a> object returned by
-          <a href="#filter-buffer-by-name-and-type"></a> algorithm with
-          <var>buffer</var> set to <a>observer buffer</a>,
-          <var>name</var> set to `null`, and <var>type</var> set
-          to `type`.</p>
-        </section>
-        <section>
-          <h2><dfn>getEntriesByName()</dfn> method</h2>
-          <p>Returns a <a>PerformanceEntryList</a> object returned by
-          <a href="#filter-buffer-by-name-and-type"></a> algorithm with
-          <var>buffer</var> set to <a>observer buffer</a>,
-          <var>name</var> set to `name`, and <var>type</var> set
-          to `null` if optional `entryType` is omitted, and <var>type</var> set
-          to `type` otherwise.</p>
-        </section>
+      <section>
+        <h2><dfn>entryTypes</dfn> attribute</h2>
+        <p>This attribute MUST return a list of entry types supported by the `PerformanceObserver` interface.</p>
+        <div class="issue">
+          <p>The above states the intent, but we need to wire up the actual plumbing to define how this ought to be implemented. To that end, a few observations..</p>
+
+          <p>This spec defines the <a>performance entry buffer</a>, which we treat as a global buffer shared by all upstream emitters and which we query via `getEntries`, `getEntriesByName`, etc. <b>However</b>, in practice this is not how UA's implement this functionality. Instead of a global buffer, all implementations maintain separate buffers (each with own size limit), which we then query depending on name/type, etc. Further, some specs <b>strong</b> define own buffer management logic (e.g. ResourceTiming).</p>
+
+          <p>We need to align our spec with implementation realities and also, ideally, define clear extension and reuse primitives to reduce duplication across specs. One way we could do this...</p>
+
+          <ul>
+            <li>Define a mechanism registry where upstream specs can register themselves with Peformance Timeline (this spec). The registration could consisnt of `name` and `size`, and Perf Timeline would then create the associated buffer and maintain it — run the add steps, enforce size constraints, etc.</li>
+            <li>Once we have the registry, we can also explain and define entryTypes — we have a list of registered names and that's what we'll return.</li>
```

Comment:
Isn't the set of supported types a static list?
Does the Custom Performance Entry proposal need registration requirement?
Could you clarify the need for dynamic registration?

Reply:
In theory, yes, it's a static list. In practice, however, different implementations enable support at different points in time, and hence we need to return a list of entry-type's _actually_ supported by the implementation to avoid [cases like this](https://github.com/w3c/performance-timeline/issues/87#issue-257140308).

Similarly, I don't want to freeze a list of strings in Perf Timeline: every time we add a new emitter, we'd have to update Perf Timeline spec, which would make things really weird and complicated for versioning. 

Last but not least, per our Blink conversation: imagine a future where certain emitters are only enabled if site is opted-in into some set of FP policies (e.g. no visited styling), that would require a further refinement in the algorithm that needs to account for what UA supports and the site policies.

Reply:
Regarding "Does the Custom Performance Entry proposal need registration requirement?" The [current proposal](https://docs.google.com/document/d/1_zm9JB-Ul_fOtAMnF6yBcmvNL3m4-k4ram4pdMIcIug/edit?pli=1#heading=h.ejti6qhmjv0b) suggests using an entryType of "custom" for all custom performance entry types, to avoid this problem.

Reply:
@tdresser right, that works. All entries generated via `queueEntry()` would have the same entryType value, but we still need some mechanism to advertise that `custom` is observable — that's the registration I'm referring to here. Roughly, in Custom Entries spec this could look something like...

> Custom Performance Entries API participates in performance timeline and registers "custom" entry type and buffer size of 100 entries.

We could define "participates and registers" formally in Perf Timeline and then use that to plumb the rest.

Reply:
Right, that sounds good to me.